src/Dictionaries/TrieDictionary.cpp

```diff
@@ -331,13 +480,136 @@ void TrieDictionary::loadData()
             {
                 const auto & attribute_column = *attribute_column_ptrs[attribute_idx];
                 auto & attribute = attributes[attribute_idx];
-                setAttributeValue(attribute, key_column->getDataAt(row_idx), attribute_column[row_idx]);
+
+                setAttributeValue(attribute, attribute_column[row]);
+            }
+
+            size_t row_number = ip_records.size();
+
+            std::string addr_str(key_column->getDataAt(row).toString());
+            size_t pos = addr_str.find('/');
+            if (pos != std::string::npos)
+            {
+                IPAddress addr(addr_str.substr(0, pos));
+                has_ipv6 = has_ipv6 || (addr.family() == Poco::Net::IPAddress::IPv6);
+
+                UInt8 prefix = std::stoi(addr_str.substr(pos + 1), nullptr, 10);
```

Comment:
Here, I suppose, we should have more user-friendly exception.
stoi may throw `std::invalid_argument`. I suppose, it should be rethrown as `DB::Exception` with key string we tried to parse in message.

Also, string like `127.0.0.1/  24something` will be successfully parsed, which is not ok

Reply:
The same, probably, should be done if we get `Poco::InvalidAddressException` from `IPAddress` constructor (if we don't have initial key string in message).